# **Import Data**

In [1]:
!kaggle competitions download -c data-analytics-competition-find-it-2024


  0%|          | 0.00/240k [00:00<?, ?B/s]
100%|██████████| 240k/240k [00:00<00:00, 291kB/s]
100%|██████████| 240k/240k [00:00<00:00, 290kB/s]


In [2]:
#import zipfile module
from zipfile import ZipFile

with ZipFile('data-analytics-competition-find-it-2024.zip', 'r') as f:
    #extract in current directory
    f.extractall()

# **Libraries**

In [4]:
import pandas as pd

# **Data Loading**

In [8]:
df_train = pd.read_csv("train_features.csv")
df_train

tahun_kelahiran pendidikan status_pernikahan   pendapatan  \
0                1979    Sarjana   Rencana Menikah          NaN   
1                1950    Sarjana   Rencana Menikah   84063000.0   
2                1966    Sarjana           Menikah  127532564.0   
3                1961   Magister   Rencana Menikah  165579620.0   
4                1970    Sarjana   Rencana Menikah  117703159.0   
...               ...        ...               ...          ...   
3812             1955   Magister           Menikah   78199470.0   
3813             1947     Doktor   Rencana Menikah  109306000.0   
3814             1974   Magister           Menikah  104621000.0   
3815             1957        SMA   Rencana Menikah  110850000.0   
3816             1961    Sarjana           Menikah  130512000.0   

      jumlah_anak_balita  jumlah_anak_remaja  terakhir_belanja  belanja_buah  \
0                    0.0                 1.0               NaN       50575.0   
1                    NaN                 NaN              70.0        6069.0   
2                    0.0                 0.0              45.0      117611.0   
3                    0.0                 0.0              90.0      206346.0   
4                    1.0                 1.0              78.0       90563.0   
...                  ...                 ...               ...           ...   
3812                 0.0                 0.0              33.0        6069.0   
3813                 0.0                 1.0              44.0           0.0   
3814                 0.0                 2.0              68.0        2023.0   
3815                 1.0                 1.0              67.0       18207.0   
3816                 0.0                 0.0              70.0      329749.0   

      belanja_daging  belanja_ikan  belanja_kue  pembelian_diskon  \
0           260967.0       50575.0      20230.0               2.0   
1            44506.0       80920.0      20230.0               9.0   
2           265460.0       96341.0     145573.0               1.0   
3          1613901.0       27725.0     125868.0               0.0   
4           311757.0       40358.0      33875.0               7.0   
...              ...           ...          ...               ...   
3812         25977.0        3856.0       5784.0               5.0   
3813         50575.0           NaN          0.0               3.0   
3814         62713.0        8092.0          0.0               7.0   
3815         70805.0       24276.0          NaN               4.0   
3816        600831.0      420784.0     161840.0               3.0   

      pembelian_web  pembelian_toko  keluhan tanggal_menjadi_anggota  
0               2.0             5.0      0.0              2014-05-05  
1               6.0             4.0      0.0              2013-03-17  
2               1.0             7.0      0.0                     NaN  
3               7.0             8.0      0.0                     NaN  
4               6.0             5.0      0.0                     NaN  
...             ...             ...      ...                     ...  
3812            1.0             0.0      0.0                     NaN  
3813            6.0             3.0      0.0              2014-06-09  
3814            5.0             7.0      0.0              2013-11-07  
3815            5.0             4.0      0.0              2013-06-30  
3816            7.0            15.0      0.0              2014-01-25  

[3817 rows x 16 columns]

**Insight :** Terdapat 3817 baris dan 16 kolom/fitur pada data train

In [9]:
df_test = pd.read_csv("test_features.csv")
df_test

ID  tahun_kelahiran pendidikan status_pernikahan   pendapatan  \
0     2241             1957    Sarjana               NaN  120660151.0   
1     2274             1968     Doktor           Menikah  163551821.0   
2     1107             1968        SMA           Menikah   29857000.0   
3     4478             1971     Doktor           Menikah  117949098.0   
4     5080             1974    Sarjana   Rencana Menikah  164761134.0   
...    ...              ...        ...               ...          ...   
3813   271             1984   Magister           Sendiri   66320000.0   
3814  3334             1972    Sarjana           Menikah  159651946.0   
3815  3478             1968   Magister   Rencana Menikah   84058923.0   
3816    89             1966    Sarjana   Rencana Menikah  106301000.0   
3817  2151             1973    Sarjana           Menikah  140353000.0   

      jumlah_anak_balita  jumlah_anak_remaja  terakhir_belanja  belanja_buah  \
0                    0.0                 1.0              63.0      122277.0   
1                    0.0                 1.0              58.0       35761.0   
2                    0.0                 0.0              34.0        8092.0   
3                    0.0                 1.0              82.0        4872.0   
4                    0.0                 0.0              28.0      343208.0   
...                  ...                 ...               ...           ...   
3813                 1.0                 0.0              19.0       12138.0   
3814                 0.0                 0.0              90.0       32119.0   
3815                 0.0                 1.0              46.0        8198.0   
3816                 0.0                 1.0              74.0       16184.0   
3817                 0.0                 0.0               9.0       62713.0   

      belanja_daging  belanja_ikan  belanja_kue  pembelian_diskon  \
0           541399.0      214192.0      84305.0               4.0   
1           353335.0       63365.0      41112.0               2.0   
2            22253.0       30345.0      26299.0               2.0   
3           126061.0           0.0       9745.0               3.0   
4          1416462.0      236196.0     107776.0               0.0   
...              ...           ...          ...               ...   
3813         38437.0           0.0      10115.0               4.0   
3814       1032587.0      176078.0      78693.0               0.0   
3815         47595.0       11392.0      10540.0               NaN   
3816         78897.0           NaN          0.0               3.0   
3817       1707412.0      453152.0          NaN               2.0   

      pembelian_web  pembelian_toko  keluhan tanggal_menjadi_anggota  
0               6.0            10.0      0.0                     NaN  
1               5.0            10.0      0.0                     NaN  
2               0.0             5.0      0.0              2013-08-06  
3               5.0             7.0      0.0                     NaN  
4               1.0             8.0      0.0                     NaN  
...             ...             ...      ...                     ...  
3813            0.0             6.0      1.0              2014-03-17  
3814            6.0             8.0      0.0                     NaN  
3815            0.0             1.0      0.0                     NaN  
3816            5.0             3.0      0.0              2014-01-20  
3817            2.0             8.0      0.0              2012-09-15  

[3818 rows x 17 columns]

**Insight :** Terdapat 3818 baris dan 17 kolom dengan tambahan fitur ID pada data test

In [10]:
df_train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3817 entries, 0 to 3816
Data columns (total 16 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   tahun_kelahiran          3817 non-null   int64  
 1   pendidikan               3628 non-null   object 
 2   status_pernikahan        3605 non-null   object 
 3   pendapatan               3627 non-null   float64
 4   jumlah_anak_balita       3627 non-null   float64
 5   jumlah_anak_remaja       3613 non-null   float64
 6   terakhir_belanja         3645 non-null   float64
 7   belanja_buah             3636 non-null   float64
 8   belanja_daging           3639 non-null   float64
 9   belanja_ikan             3624 non-null   float64
 10  belanja_kue              3603 non-null   float64
 11  pembelian_diskon         3639 non-null   float64
 12  pembelian_web            3652 non-null   float64
 13  pembelian_toko           3648 non-null   float64
 14  keluhan                 

**Insight :** 
1. Dengan jumlah baris 3817 pada data train, semua memiliki missing value terkecuali fitur tahun_kelahiran
2. Fitur tanggal_menjadi_anggota seharusnya memiliki tipe data datetime

In [11]:
df_test.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3818 entries, 0 to 3817
Data columns (total 17 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   ID                       3818 non-null   int64  
 1   tahun_kelahiran          3818 non-null   int64  
 2   pendidikan               3614 non-null   object 
 3   status_pernikahan        3636 non-null   object 
 4   pendapatan               3615 non-null   float64
 5   jumlah_anak_balita       3609 non-null   float64
 6   jumlah_anak_remaja       3608 non-null   float64
 7   terakhir_belanja         3617 non-null   float64
 8   belanja_buah             3632 non-null   float64
 9   belanja_daging           3623 non-null   float64
 10  belanja_ikan             3622 non-null   float64
 11  belanja_kue              3631 non-null   float64
 12  pembelian_diskon         3639 non-null   float64
 13  pembelian_web            3638 non-null   float64
 14  pembelian_toko          

**Insight :** 
1. Hampir sama dengan data train, hanya fitur ID dan tahun_kelahiran yang tidak memiliki missing value
2. Fitur tanggal_menjadi_anggota seharusnya memiliki tipe data datetime

In [12]:
df_train.describe()

tahun_kelahiran    pendapatan  jumlah_anak_balita  jumlah_anak_remaja  \
count      3817.000000  3.627000e+03         3627.000000         3613.000000   
mean       1967.823946  1.144832e+08            0.293080            0.353723   
std          11.768131  4.346042e+07            0.473063            0.493014   
min        1899.000000  5.073000e+06            0.000000            0.000000   
25%        1959.000000  8.112512e+07            0.000000            0.000000   
50%        1968.000000  1.156214e+08            0.000000            0.000000   
75%        1976.000000  1.504960e+08            1.000000            1.000000   
max        2000.000000  3.328840e+08            2.000000            2.000000   

       terakhir_belanja   belanja_buah  belanja_daging   belanja_ikan  \
count       3645.000000    3636.000000    3.639000e+03    3624.000000   
mean          47.231550   59804.239824    4.385748e+05   81428.997792   
std           27.068512   74024.976109    5.120427e+05   99976.226855   
min            0.000000       0.000000    0.000000e+00       0.000000   
25%           25.000000    7907.000000    4.947950e+04   10115.000000   
50%           47.000000   26456.000000    2.219930e+05   36054.500000   
75%           69.000000   86162.000000    6.863555e+05  121380.000000   
max          128.000000  396508.000000    3.489675e+06  621600.000000   

         belanja_kue  pembelian_diskon  pembelian_web  pembelian_toko  \
count    3603.000000       3639.000000    3652.000000     3648.000000   
mean    63377.970580          2.125584       4.436473        5.767818   
std     79435.457282          2.100133       3.002522        3.210738   
min         0.000000          0.000000       0.000000        0.000000   
25%      7947.000000          0.000000       2.000000        3.000000   
50%     27795.000000          2.000000       4.000000        5.000000   
75%     89502.500000          3.000000       7.000000        8.000000   
max    542164.000000         20.000000      30.000000       17.000000   

           keluhan  
count  3621.000000  
mean      0.004971  
std       0.070340  
min       0.000000  
25%       0.000000  
50%       0.000000  
75%       0.000000  
max       1.000000

In [13]:
df_test.describe()

ID  tahun_kelahiran    pendapatan  jumlah_anak_balita  \
count  3818.000000      3818.000000  3.615000e+03         3609.000000   
mean   3841.507072      1967.789942  1.163249e+08            0.282904   
std    2198.401006        11.808572  5.105122e+07            0.465005   
min       2.000000      1892.000000  2.998899e+06            0.000000   
25%    1950.750000      1959.000000  8.168162e+07            0.000000   
50%    3872.500000      1968.000000  1.177321e+08            0.000000   
75%    5770.750000      1976.000000  1.506877e+08            1.000000   
max    7634.000000      2009.000000  1.305740e+09            2.000000   

       jumlah_anak_remaja  terakhir_belanja   belanja_buah  belanja_daging  \
count         3608.000000       3617.000000    3632.000000    3.623000e+03   
mean             0.355322         47.181366   61234.083425    4.528484e+05   
std              0.496865         26.940287   76647.590617    5.123206e+05   
min              0.000000          0.000000       0.000000    0.000000e+00   
25%              0.000000         25.000000    7497.500000    5.462100e+04   
50%              0.000000         46.000000   29525.000000    2.327530e+05   
75%              1.000000         68.000000   86776.750000    7.217960e+05   
max              2.000000        133.000000  408646.000000    3.271191e+06   

        belanja_ikan    belanja_kue  pembelian_diskon  pembelian_web  \
count    3622.000000    3631.000000       3639.000000    3638.000000   
mean    85338.612093   63801.505921          2.101676       4.465091   
std    103789.546199   78428.544713          2.115571       2.935615   
min         0.000000       0.000000          0.000000       0.000000   
25%     10115.000000    8092.000000          0.000000       2.000000   
50%     40025.500000   29251.000000          2.000000       4.000000   
75%    129656.750000   91407.000000          3.000000       6.000000   
max    574932.000000  532049.000000         17.000000      27.000000   

       pembelian_toko      keluhan  
count     3632.000000  3625.000000  
mean         5.835628     0.002759  
std          3.288349     0.052457  
min          0.000000     0.000000  
25%          3.000000     0.000000  
50%          5.000000     0.000000  
75%          8.000000     0.000000  
max         17.000000     1.000000

**Insight :** 
1. Hampir semua fitur memiliki kemungkinan distribusi normal dikarenakan memiliki perbedaan nilai yang tidak terlalu jauh
2. Fitur fitur belanja yang mengenai harga memiliki kemungkinan outlier dan distribusinya skew

In [14]:
df_train.duplicated().sum()

0

In [15]:
df_test.duplicated().sum()

0

# **EDA**